server/utils/utils.ts

```diff
@@ -51,3 +51,36 @@ export function mapDeliusAddressToBreachNoticeAddress(deliusAddress: DeliusAddre
   }
   return null
 }
+
+export function formatAddressForSelectMenuDisplay(deliusAddress: DeliusAddress): string {
+  const formattedAddressArray: string[] = []
+  if (deliusAddress) {
+    if (deliusAddress.buildingName) {
+      formattedAddressArray.push(deliusAddress.buildingName)
+    }
+    if (deliusAddress.buildingNumber || deliusAddress.streetName) {
+      const buildingNoStreetNameArray: string[] = []
+      if (deliusAddress.buildingNumber) {
+        buildingNoStreetNameArray.push(deliusAddress.buildingNumber)
+      }
+      if (deliusAddress.streetName) {
+        buildingNoStreetNameArray.push(deliusAddress.streetName)
+      }
+      formattedAddressArray.push(buildingNoStreetNameArray.join(' '))
+    }
+    if (deliusAddress.district) {
+      formattedAddressArray.push(deliusAddress.district)
+    }
+    if (deliusAddress.townCity) {
+      formattedAddressArray.push(deliusAddress.townCity)
+    }
+    if (deliusAddress.county) {
+      formattedAddressArray.push(deliusAddress.county)
+    }
+    if (deliusAddress.postcode) {
+      formattedAddressArray.push(deliusAddress.postcode)
+    }
+    return formattedAddressArray.join(', ')
+  }
+  return null
+}
```

Comment:
Was the problem in the previous approach to do with buildingNumber / streetName - as it looks like they weren't null-checked?  Is there any reason not to keep it simple and just apply the same filter logic to those two fields?

```suggestion
export function formatAddressForSelectMenuDisplay(deliusAddress: DeliusAddress): string {
  if (deliusAddress) {
    return [
      address.buildingName,
      [address.buildingNumber, address.streetName].filter(item => item).join(' ').trim(),
      address.district,
      address.townCity,
      address.county,
      address.postcode,
    ].filter(item => item).join(', '),
  }
  return null
}
```